plugins/techdocs/src/client.ts

```diff
@@ -192,44 +193,59 @@ export class TechDocsStorageClient implements TechDocsStorageApi {
    * Check if docs are on the latest version and trigger rebuild if not
    *
    * @param {EntityName} entityId Object containing entity data like name, namespace, etc.
+   * @param {Function} logHandler Callback to receive log messages from the build process
    * @returns {SyncResult} Whether documents are currently synchronized to newest version
    * @throws {Error} Throws error on error from sync endpoint in Techdocs Backend
    */
-  async syncEntityDocs(entityId: EntityName): Promise<SyncResult> {
+  async syncEntityDocs(
+    entityId: EntityName,
+    logHandler: (line: string) => void = () => {},
+  ): Promise<SyncResult> {
     const { kind, namespace, name } = entityId;
 
     const apiOrigin = await this.getApiOrigin();
     const url = `${apiOrigin}/sync/${namespace}/${kind}/${name}`;
     const token = await this.identityApi.getIdToken();
-    let request;
-    let attempts: number = 0;
-    // retry if request times out, up to 5 times
-    // can happen due to docs taking too long to generate
-    while (!request || (request.status === 408 && attempts < 5)) {
-      attempts++;
-      request = await fetch(url, {
+
+    return new Promise((resolve, reject) => {
+      const source = new EventSource(url, {
+        withCredentials: true,
```

Comment:
There are some users who use this [authenticating middleware](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md) and there have been [repeating discussions](https://github.com/backstage/backstage/issues/5041) about it. The scaffolder for example [uses this](https://github.com/backstage/backstage/blob/cecfe5c7f5a65bba79520146008833176930ba92/plugins/scaffolder/src/api.ts#L211) because one can't add headers to the original `EventSource`. We don't use it and thus I don't know what they really need, but I also don't want to break it for those users. Maybe @benjdlambert can give some input on what to do in this situation.